src/compiler/checker.ts

```diff
@@ -4414,8 +4414,10 @@ namespace ts {
                     }
                 }
             }
+        }
 
-            if (hasQuestionToken(node)) {
+        function isOptionalParameter(node: ParameterDeclaration) {
+            if (isJSDocOptionalParameter(node) || hasQuestionToken(node)) {

```

Comment:
Maybe. Isn't `flags & NodeFlags.JavaScriptFile` faster than `switch(node.kind)` with 7 cases? In other words, I think the speed gain will be absolutely tiny. And the code was originally written in this order, although that's (apparently) not meaningful.